docs/concepts/flows.md

```diff
@@ -1,44 +1,67 @@
-[Flowable Tasks](../developer-guide/flowable) are tasks that only handle workflow states and trigger new tasks. They will handle the logic of the flow enabling complex workflow like branching, parallel tasks, etc...  
-An example is a `Switch` task that can take any inputs and decide what are the next tasks to run (and to allow branching). 
+## Task
 
-A Flowable Task is handled by `Executors` and can be called very often, so these tasks cannot include intensive computations.
+A task is a single action in a flow. A task can take properties as inputs, perform an action, and produce an output.
+
+There are two kinds of tasks in Kestra:
+- Runnable Task 
+- Flowable Task
 
-The most common Flowable Tasks will be available in the core. 
 
 ### Runnable Task 
-Runnable Tasks are used to handle any computational work. This can be anything, including file system operations, api calls, database queries, and more. These tasks can be compute intensive so they will be handled by `Workers`. 
 
-By default, Kestra does not include a lot of Runnable Tasks, most will be available as Plugins.
+Runnable Tasks handle computational work in the flow. For example, file system operations, API calls, database queries, etc. These tasks can be compute-intensive and are handled by [workers](../architecture/#worker). 
+
+By default, Kestra only includes a few Runnable Tasks. However, many of them are available as [plugins](../../plugins/).
+
+
+### Flowable Task
+
+[Flowable Tasks](../developer-guide/flowable) only handle flow logic (branching, grouping, parallel processing, etc.) and trigger new tasks. For example, the Switch task takes inputs and decides the next tasks to run. 
+
+A Flowable Task is handled by executors and can be called very often. Because of that, these tasks cannot include intensive computations, unlike Runnable Tasks. Most of the common Flowable Tasks are available in the default Kestra installation. 
+
 
 ## Namespace
-A Namespace is like a file systems folder, it aims to group flows in a limitless hierarchy.
-A Namespace is composed of word and letters separated by `.`. The hierarchy depth is not limited and is free for Kestra users to modify to fit their organization.
+
+A namespace is like a folder for flows. A namespace is composed of words and letters separated by `.`. The hierarchy depth for Namespaces is unlimited. Here are some examples of namespaces:
+- `projectOne`
+- `com.projectTwo`
+- `test.projectThree.folder`
+
 
 ## Inputs 
-[Inputs](../developer-guide/inputs) are parameters sent to a flow at execution time. It can be anything (string, int, file, ...) that will be sent when the execution is created. The Flow must define possible inputs, and mandatory ones are enforced before the creation of the execution. 
+
+[Inputs](../developer-guide/inputs) are parameters sent to a flow at execution time. It can be anything, for example, a string, integer, file, etc. The flow must define any possible inputs before its execution. 
```

Comment:
>The flow must define any possible inputs before its execution. 

The idea behind this sentence is that all inputs must be set before the flow execution, I think it should be rephrased

Reply:
Doesn't it resolve to the same meaning?

Reply:
I understand what it means now. Fixed :)